### PR TITLE
feat: get latest anchor origin

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: component/vdr/orb/config/service.go [hunk 1]
```diff
@@ -337,10 +338,61 @@ func (cs *Service) getEndpointIPNS(didURI string) (*models.Endpoint, error) {
...
 
 	return cs.populateResolutionEndpoint(fmt.Sprintf("%s/%s/%s", ipfsGlobal, "ipns", anchorOriginSplit[1]),
-		url.PathEscape(anchorOrigin))
+		url.PathEscape(currentAnchorOrigin))
+}
+
+func (cs *Service) getLatestCID(anchorOrigin, suffix string) (string, error) {
+	var webFingerResponse restapi.WebFingerResponse
+
+	anchorOriginSplit := strings.Split(anchorOrigin, "ipns://")
```
Comment (troyronda) [inline component/vdr/orb/config/service.go:365]: This also needs to work for https origins.

Comment (fqutishat) [reply component/vdr/orb/config/service.go]: @troyronda done

File: component/vdr/orb/config/service.go [hunk 2]
```diff
@@ -333,14 +333,85 @@ func (cs *Service) getEndpointIPNS(didURI string) (*models.Endpoint, error) {
 		return nil, fmt.Errorf("get anchor origin didn't return string")
 	}
 
-	if !strings.HasPrefix(anchorOrigin, "ipns://") {
-		return nil, fmt.Errorf("anchor origin is not ipns")
+	currentAnchorOrigin := anchorOrigin
+
+	for {
+		latestAnchorOrigin, errGet := cs.getLatestCID(currentAnchorOrigin, didSplit[4])
```
Comment (sandrask) [inline component/vdr/orb/config/service.go:339]: function should be called getLatestAnchorOrigin

**Review by troyronda — APPROVED** [review]

File: component/vdr/orb/config/service.go (hunk 2 above)
Comment (fqutishat) [reply component/vdr/orb/config/service.go]: done

**Review by sandrask — APPROVED** [review]

**Pull request merged**